yapapi/runner/__init__.py

```diff
@@ -204,6 +204,36 @@ def on_work_done(task, status):
         workers: Set[asyncio.Task[None]] = set()
         last_wid = 0
 
+        agreements_to_pay: Set[str] = set()
+        invoices: Dict[str, rest.payment.Invoice] = dict()
+        payment_closing: bool = False
+
+        async def process_invoices():
+            assert self._budget_allocation
+            allocation: rest.payment.Allocation = self._budget_allocation
+            async for invoice in self._payment_api.incoming_invoices():
+                if invoice.agreement_id in agreements_to_pay:
+                    agreements_to_pay.remove(invoice.agreement_id)
+                    await invoice.accept(amount=invoice.amount, allocation=allocation)
+                else:
+                    invoices[invoice.agreement_id] = invoice
+                if payment_closing and not agreements_to_pay:
+                    break
+
+        async def accept_payment_for_agreement(agreement_id: str, *, partial=False) -> bool:
```

Comment:
`partial` parameter seems to be never used.

Reply:
it distinguishes two calls below and as such improves readability there